### PR TITLE
Advanced filtering

### PR DESCRIPTION
**Pull request by gsfr**
@rentzso LGTM. Are you ok to merge?

### CODE REVIEW

File: api/sessions.py [hunk 1]
```diff
@@ -115,11 +115,19 @@ def get(self, pid=None, gid=None):
...
+            session_measurements = self.app.db.acquisitions.aggregate([
+                {'$match': {'session': {'$in': [sess['_id'] for sess in sessions]}}},
+                {'$group': {'_id': '$session', 'measurements': {'$addToSet': '$datatype'}}}
+                ])
+            session_measurements = {sess['_id']: sess['measurements'] for sess in session_measurements}
         for sess in sessions:
+            sess['measurements'] = session_measurements.get(sess['_id'], None)
             if 'subject_code' not in sess:
-                sess['subject_code'] = sess.pop('subject', {}).get('code', '') # FIXME when subject is pulled out of session
+                sess['subject_code'] = sess.get('subject', {}).get('code', '') # FIXME when subject is pulled out of session
```
Comment (kofalt) [inline api/sessions.py]: Was `pop` the correct approach before this change?

File: api/sessions.py [hunk 2]
```diff
@@ -115,11 +115,19 @@ def get(self, pid=None, gid=None):
...
         sessions = self._get(query, projection, self.request.GET.get('admin', '').lower() in ('1', 'true'))
+        session_measurements = {}
+        if self.request.GET.get('measurements', '').lower() in ('1', 'true'):
+            session_measurements = self.app.db.acquisitions.aggregate([
+                {'$match': {'session': {'$in': [sess['_id'] for sess in sessions]}}},
+                {'$group': {'_id': '$session', 'measurements': {'$addToSet': '$datatype'}}}
+                ])
+            session_measurements = {sess['_id']: sess['measurements'] for sess in session_measurements}
         for sess in sessions:
+            sess['measurements'] = session_measurements.get(sess['_id'], None)
```
Comment (kofalt) [inline api/sessions.py]: To me, it looks like this line will overwrite any pre-existing `measurements` key on the object if `measurements` was not passed as a URL param. Is there any case where that might be a problem?

Comment (rentzso) [inline api/sessions.py]: 'measurements' isn't a key on session objects.

File: api/sessions.py (hunk 1 above)
Comment (rentzso) [inline api/sessions.py]: get is probably better :)

File: api/sessions.py (hunk 2 above)
Comment (kofalt) [inline api/sessions.py]: Cool, just checking.

File: api/sessions.py (hunk 1 above)
Comment (kofalt) [inline api/sessions.py]: Haha, okay :)

**Pull request merged**